operations/service_deploy_smoke_utils.go

```diff
@@ -197,21 +161,23 @@ func checkHostTaskByCommit(username, key string) error {
 		}
 		break
 	}
+	return checkForTask(client, agent.HostMode, builds[0].Tasks, username, key)
+}
 
-	var task apimodels.APITask
+func checkForTask(client *http.Client, mode agent.Mode, tasks []string, username, key string) error {
+	var taskStatus string
 OUTER:
 	for i := 0; i <= 30; i++ {
 		// check task
 		if i == 30 {
-			return errors.Errorf("task status is %s (expected %s)", task.Status, evergreen.TaskSucceeded)
+			return errors.Errorf("task status is %s (expected %s)", taskStatus, evergreen.TaskSucceeded)
 		}
 		time.Sleep(10 * time.Second)
-		grip.Infof("checking for %d tasks (%d/30)", len(builds[0].Tasks), i+1)
+		grip.Infof("checking for %d tasks (%d/30)", len(tasks), i+1)
 
-		var err error
-		for t := 0; t < len(builds[0].Tasks); t++ {
-			taskId := builds[0].Tasks[t]
-			task, err = checkTask(client, username, key, taskId)
+		for t := 0; t < len(tasks); t++ {
```

Comment:
nit: this could be re-written to a `for _, taskId := range tasks` since it doesn't use the index for anything else.